src/v/cloud_topics/level_one/metastore/metastore.h

```diff
@@ -294,6 +295,47 @@ class metastore {
       const model::topic_id_partition&,
       model::timestamp tombstone_removal_upper_bound_ts)
       = 0;
+
+    // All the information required to query a `compaction_info_response` from
+    // the metastore. Parameters are used for call to
+    // `get_compaction_offsets()`.
+    struct sample_spec {
```

Comment:
nit: would appreciate if the name reflected that this is used for compaction only, like compaction_sample_spec or compaction_info_request.